### PR TITLE
Log feature importance, add internal use metadata api

### PR DESCRIPTION
**Pull request by yuhong-sigopt**

### CODE REVIEW

File: sigopt/xgboost/run.py [hunk 1]
```diff
@@ -154,6 +154,15 @@ def train_xgb(self):
       if self.run_options_parsed['log_stderr']:
         log_dict["stderr"] = stderr
       self.run.set_logs(log_dict)
+
+    scores = bst.get_score(importance_type='weight', fmap='')
+    scores = dict(sorted(scores.items(), key=lambda x:-x[1])[:FEATURE_IMPORTANCE_MAX_NUM_FEATURE])
```
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/run.py]: `scores[:M] is equal to score[:min(len(scores), M)]`. 

`reverse=True` is changed.

File: integration_test/test_xgboost.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
   },
   dtrain=D_train,
   evals=[(D_test, 'test0'), (D_test, 'test1')],
-  num_boost_round=10,
+  num_boost_round=3,
 )
 
 class TestXGBoost(object):
```
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply integration_test/test_xgboost.py]: Done.

**Review by ericlee0803 — APPROVED** [review]
the addition of feature importance inside xgboost/runs.py and associated test looks fine to me, I defer to others as far as the rest of the code is concerned.

**Comment by ericlee0803** [comment]
@yuhong-sigopt can you rebase? Just merged in a PR for the metrics logging

**Review by ericlee0803 — CHANGES_REQUESTED** [review]
rebase off metric logging PR

**Comment by yuhong-sigopt** [comment]
> @yuhong-sigopt can you rebase? Just merged in a PR for the metrics logging

Done.

**Comment by yuhong-sigopt** [comment]
> rebase off metric logging PR

Done.

**Review by ericlee0803 — APPROVED** [review]

**Pull request merged**